jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/Server.java

```diff
@@ -161,6 +162,13 @@ public void handle(Request request)
         }
     }
 
+    @Override
+    public InvocationType getInvocationType()
+    {
+        // Return cached type to avoid a full handler tree walk.
+        return _invocationType;
+    }
+
```

Comment:
We really want to avoid traversing the handler tree just to work out the type.    But if we do decide to use a two phase accept/process approach, then it is possible that the accept phase could return the type of the process phase.   But that does us no good if the accept phase blocks, so we'd still need handlers to declare if their accept phase blocks or not.